packages/twenty-server/src/workspace/cron/clean-inactive-workspaces/clean-inactive-workspace.job.ts

```diff
@@ -196,29 +207,94 @@ export class CleanInactiveWorkspaceJob implements MessageQueueJob<undefined> {
     );
   }
 
-  async handle(): Promise<void> {
-    this.logger.log('Job running...');
+  getDryRunLog(isDryRun: boolean): string {
+    return isDryRun ? 'Dry-run mode: ' : '';
+  }
+
+  chunkArray(array: any[], chunkSize = 6): any[][] {
+    const chunkedArray: any[][] = [];
+    let index = 0;
+
+    while (index < array.length) {
+      chunkedArray.push(array.slice(index, index + chunkSize));
+      index += chunkSize;
+    }
+
+    return chunkedArray;
+  }
+
+  async sendDeleteWorkspaceEmail(isDryRun: boolean): Promise<void> {
+    if (isDryRun || this.workspacesToDelete.length === 0) {
+      return;
+    }
+    const emailTemplate = DeleteInactiveWorkspaceEmail(this.workspacesToDelete);
+    const html = render(emailTemplate, {
+      pretty: true,
+    });
+    const text = render(emailTemplate, {
+      plainText: true,
+    });
+
+    await this.emailService.send({
+      to: this.environmentService.getEmailSystemAddress(),
+      from: `${this.environmentService.getEmailFromName()} <${this.environmentService.getEmailFromAddress()}>`,
+      subject: 'Action Needed to Delete Workspaces',
+      html,
+      text,
+    });
+  }
+
+  async handle(data: CleanInactiveWorkspacesCommandOptions): Promise<void> {
+    const isDryRun = data.dryRun || false;
+
+    this.logger.log(`${this.getDryRunLog(isDryRun)}Job running...`);
     if (!this.inactiveDaysBeforeDelete && !this.inactiveDaysBeforeEmail) {
       this.logger.log(
         `'WORKSPACE_INACTIVE_DAYS_BEFORE_NOTIFICATION' and 'WORKSPACE_INACTIVE_DAYS_BEFORE_DELETION' environment variables not set, please check this doc for more info: https://docs.twenty.com/start/self-hosting/environment-variables`,
       );
 
       return;
     }
+
     const dataSources =
       await this.dataSourceService.getManyDataSourceMetadata();
```

Comment:
I wonder if chuncking the dataSources when we have them is enough. We might need to actually query the datasources by chunk (using SQL LIMIT and OFFSET)

Reply:
ok

Reply:
seen together, it should work without chunking dataSources as it is not that much data